linked_list/implementation.py

```diff
@@ -1,40 +1,91 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
+from functools import reduce
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
-    def __init__(self, elements=None):
-        pass
+    def __init__(self, elements=[]):
+        #self.elements = elements
+        self.start = None
+        self.end = None
+        for elem in elements:
+            self.append(elem)
 
     def __str__(self):
-        pass
+        return_string = "["
+        for node in self:
+            return_string += str(node)
+            if not node is self.end:
+                return_string += ", "
+        return_string += "]"
+        return return_string
 
     def __len__(self):
-        pass
-
+        return reduce(lambda x,y: x+1, [node for node in self], 0)
+                
     def __iter__(self):
```

Comment:
Great use of generators! It saves you from implementing the next() method and lets you avoid declaring an extra class attribute to keep track of which element you're on.